tests/shared/src/test/scala/cats/effect/IOSpec.scala

```diff
@@ -1342,6 +1342,50 @@ class IOSpec extends BaseSpec with Discipline with IOPlatformSpecification {
             case Right(_) => SyncIO.raiseError[Unit](new RuntimeException("Boom!"))
           } must completeAsSync(())
       }
+
+      "should not execute side-effects twice for map (#2858)" in ticked { implicit ticker =>
```

Comment:
```suggestion
      "should not execute effects twice for map (#2858)" in ticked { implicit ticker =>
```

Reading this the first time, I thought from the description that this was about shapes like `io.map(_ => i += 1)`

Reply:
Oh yeah, that's unclear. It's observed as the side-effects being run twice, but it is in fact the entire effect being run twice. Thanks!